### PR TITLE
change: cache supported ops in the device

### PR DESCRIPTION
**Pull request by antalszava**
*Description of changes:*

The `operations` property of a device returns the set of operations supported by the device. As the operations set is computed on every call, the runtime is non-negligible when called many times e.g., during an optimization.

This PR adds caching for the operations so that the set is computed only once.

*Note*: the PR branches off of #62, so some older commits are also shown as added commits.

*Testing done:*

Added a unit test that checks that the new `_supported_ops` internal attribute stores the supported operations when the `operations` property is used.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by licedric — APPROVED** [review]
Nice catch and thanks!

File: src/braket/pennylane_plugin/braket_device.py [hunk 1]
```diff
@@ -111,7 +112,9 @@ def capabilities(cls):
     @property
     def operations(self) -> FrozenSet[str]:
         """FrozenSet[str]: The set of names of PennyLane operations that the device supports."""
-        return supported_operations()
+        if self._supported_ops is None:
+            self._supported_ops = supported_operations()
+        return self._supported_ops
```
Comment (speller26) [inline src/braket/pennylane_plugin/braket_device.py:117]: Why not just have `self._supported_ops = supported_operations()` in `__init__`? Then this just becomes
```python
return self._supported_ops
```

Comment (antalszava) [reply src/braket/pennylane_plugin/braket_device.py]: Good point, updated! Thanks

**Comment by antalszava** [comment]
@licedric thank you for the review! Updated as per @speller26's suggestion, it seems that removed your previous review :thinking:

**Review by speller26 — APPROVED** [review]

**Review by licedric — APPROVED** [review]

**Pull request merged**